Bio/SeqIO/Interfaces.py

```diff
@@ -39,39 +42,39 @@ def __init__(self, source, alphabet=None, mode="t", fmt=None):
         - you do not have to require an alphabet.
         - you can add additional optional arguments.
         """
+        self.source = source
         if alphabet is not None:
             raise ValueError("The alphabet argument is no longer supported")
         try:
-            self.stream = open(source, "r" + mode)
-            self.should_close_stream = True
+            self.stream = open(source, "r" + self.mode)
+            # self.should_close_stream = True
         except TypeError:  # not a path, assume we received a stream
-            if mode == "t":
+            if self.mode == "t":
                 if source.read(0) != "":
                     raise StreamModeError(
-                        f"{fmt} files must be opened in text mode."
+                        f"{self.fmt} files must be opened in text mode."
                     ) from None
             elif mode == "b":
                 if source.read(0) != b"":
                     raise StreamModeError(
-                        f"{fmt} files must be opened in binary mode."
+                        f"{self.fmt} files must be opened in binary mode."
                     ) from None
             else:
-                raise ValueError(f"Unknown mode '{mode}'") from None
+                raise ValueError(f"Unknown mode '{self.mode}'") from None
             self.stream = source
-            self.should_close_stream = False
+            # self.should_close_stream = False
         try:
             self.records = self.parse(self.stream)
         except Exception:
-            if self.should_close_stream:
-                self.stream.close()
-            raise
+            self.stream.close()
+        raise
```

Comment:
This raise is not indented as part of the exception block, thus errors like this from the tests:

```
2023-05-07T22:11:16.9745569Z ----------------------------------------------------------------------
2023-05-07T22:11:16.9745962Z File "/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/site-packages/Bio/SeqRecord.py", line 1041, in Bio.SeqRecord.SeqRecord.lower
2023-05-07T22:11:16.9746052Z Failed example:
2023-05-07T22:11:16.9746184Z     old = SeqIO.read("EMBL/TRBG361.embl", "embl")
2023-05-07T22:11:16.9746274Z Exception raised:
2023-05-07T22:11:16.9746383Z     Traceback (most recent call last):
2023-05-07T22:11:16.9746589Z       File "/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/doctest.py", line 1336, in __run
2023-05-07T22:11:16.9746729Z         exec(compile(example.source, filename, "single",
2023-05-07T22:11:16.9746918Z       File "<doctest Bio.SeqRecord.SeqRecord.lower[5]>", line 1, in <module>
2023-05-07T22:11:16.9747054Z         old = SeqIO.read("EMBL/TRBG361.embl", "embl")
2023-05-07T22:11:16.9747400Z       File "/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/site-packages/Bio/SeqIO/__init__.py", line 652, in read
2023-05-07T22:11:16.9747536Z         iterator = parse(handle, format, alphabet)
2023-05-07T22:11:16.9747876Z       File "/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/site-packages/Bio/SeqIO/__init__.py", line 605, in parse
2023-05-07T22:11:16.9748097Z         return iterator_generator(handle)
2023-05-07T22:11:16.9748449Z       File "/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/site-packages/Bio/SeqIO/InsdcIO.py", line 151, in __init__
2023-05-07T22:11:16.9748585Z         super().__init__(source, mode="t", fmt="EMBL")
2023-05-07T22:11:16.9748938Z       File "/opt/hostedtoolcache/Python/3.8.16/x64/lib/python3.8/site-packages/Bio/SeqIO/Interfaces.py", line 70, in __init__
2023-05-07T22:11:16.9749015Z         raise
2023-05-07T22:11:16.9749147Z     RuntimeError: No active exception to reraise
2023-05-07T22:11:16.9749373Z ----------------------------------------------------------------------
```